includes/transforms/ExcludeDirectories.h

```diff
@@ -0,0 +1,33 @@
+#ifndef NSFW_TRANSFORMS_EXCLUDEDIRECTORIES_H
+#define NSFW_TRANSFORMS_EXCLUDEDIRECTORIES_H
+
+#include "AbstractTransform.h"
+
+#include <regex>
+
+class ExcludeDirectories : public AbstractTransform
+{
+public:
+  ExcludeDirectories(const std::string &regex)
+    : regex(regex) {}
+
+  VecEvents transform(VecEvents vecEvents) {
+    vecEvents->erase(std::remove_if(vecEvents->begin(),
+                                 vecEvents->end(),
+                                 [this](std::unique_ptr<Event> &event){
+      std::regex self_regex(regex,
+                            std::regex_constants::ECMAScript
+                          | std::regex_constants::icase);
```

Comment:
not sure about the detailed semantics here. Though, my feeling is that you should set up `self_regex` outside the lambda and capture it. Otherwise it would recompile the regex several times, no?